Add arrays support for deferred chips

victordidenko opened the pull request:
This pull-request is similar to #36, but adds also arrays support for deferred chips.
When user enters text -- render function performs request to server and adds all returned results to chips.

victordidenko committed:
Add arrays support for deferred chips

mohbasheer commented:
Thanks for the Pull Request.

Could you please explain me the use case like when its required.

victordidenko commented:
As I wrote -- when user enters text, render function performs request to server and adds all returned results to chips. If user don't remember exact name of a chip (or whatever it is in the interface), but remembers part of it. And after that user can remove unnecessary items.
![array](https://user-images.githubusercontent.com/15016227/33379087-15829fe6-d528-11e7-9172-a7224bb9d0ed.gif)

victordidenko commented:
It is possible to achieve that by modifying model value, but in that case red items ("qwer" in gif above) will disappear. So we need support from library here.

mohbasheer commented:
I got it.  Could you add sample implementation of the same in samples. It will help me lot.

victordidenko committed:
Add example for arrays / Fixed chips deletion

victordidenko commented:
Sure, added "Using Promise, returning array (with list of string)" to examples.
You can type "korea" to get both Koreas at once, or "republic" to get all Republics.
Also, fixed deletion of chips.
